operator-common/src/main/java/io/strimzi/operator/common/OperatorKubernetesClientBuilder.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+
+/**
+ * Class for generating Kubernetes Clients for Operators.
+ */
+public class OperatorKubernetesClientBuilder {
```

Comment:
This looks good. But would a single static method which is just called be enough instead of having a full class for it? You can for example name it `KubernetesClientUtils` and call the static method `clientWithUserAgent` or something.

WDYT @tombentley @ppatierno?

Reply:
I'm not used to Strimzi's coding style, I didn't know what your opinion on statics and *Util classes are, and I copied the style from PasswordGenerator. Maybe that was a bad model for this.

Let me know if you prefer the Utils class with the static method and I can switch to it.

Reply:
Tbh I like the idea of having the same approach as the `KubernetesClientBuilder` so I am fine with the way it's done now. But it's not a strong opinion, if others prefer the Util classes I am fine too.

Reply:
Ok, if Paolo is fine with it lets stick with it. So I guess its just the Javadoc comment he left. Thanks for your patience.